File: src/main/java/linq/exception/ThrowHelper.java
package linq.exception;

public class ThrowHelper {
    public static void throwArgumentNullException(ExceptionArgument argument) {
        throw new NullPointerException(argument.name() + " is null");
    }

    public static void throwNoElementsException() {
        throw new IllegalStateException(SR.NoElements);
    }

    public static void throwNoMatchException() {
        throw new IllegalStateException(SR.NoMatch);
    }
}

File: src/main/kotlin/linq/exception/ThrowHelper.kt
package linq.exception

fun throwArgumentNullException(argument: ExceptionArgument): Nothing {
    throw NullPointerException(argument.string + " is null")
}

fun throwArgumentOutOfRangeException(argument: ExceptionArgument): Nothing {
    throw IllegalArgumentException(argument.string + " is out of range")
}

fun throwNoElementsException(): Nothing {
    throw IllegalStateException(SR.NO_ELEMENTS)
}

fun throwNoMatchException(): Nothing {
    throw IllegalStateException(SR.NO_MATCH)
}